File: java/FRead.java
/*
Read A File 

Steps
-----
* open the file
* fetch data until EOF
  - print the fetched data
*close the file

*/

import java.io.*;

class FRead
{
 public static void main(String args[])
 {
  try
  {
   String fname = args[0];

   //open the file
   FileInputStream fin = new FileInputStream(fname);//file must exist   

   //read file data   
   int data;
   int cnt = 0;
  
   while( (data = fin.read()) != -1)//-1 means EOF
   {
     System.out.write(data);
     cnt++;
   } 
   System.out.flush();//ensure buffered data is transmitted to System.out

   //close the file
   fin.close();

   System.out.println("\n File Read in "+ cnt + " cycles ");
  }
  catch(ArrayIndexOutOfBoundsException ex)
  {
   System.out.println("Usage : java FRead <fileName>");
  }
  catch(IOException ex)
  {
   System.out.println(ex);
  }

 }//main
}//FRead

File: C++ Notes/fread.cpp
#include <iostream>
#include <fstream>

using namespace std;
/*
File Handling
--------------
File Handling means performing input(read)
and output(write) operations on files that
are stored on disk.

General Steps for File Handling
--------------------------------
1) Open the file
2) Read/Write
3) Close the file


C++ provides numerous classes and function
(through header file fstream.h) that help
in performing file operations.
*/

//Read A File

int main()
{
	//1. create object of ifstream
	ifstream fin;
	
	//2. open the file
	fin.open("d:/a.txt");//file must exist
	
	//3. test open status
	if(fin)
	{
		//4. reading
		int data;
		
		while((data = fin.get()) != EOF)
		{
			cout<<(char)data;
		}
		//5. close
		fin.close();		
	}	
	else
		cout<<"\n File Open Failed";
		
	return 0;
}

/*
ifstream : a class in fstream.h that support file reading
open : a function of ifstream class that is used to open a file.
get() : a function of ifstream class that is used to fetch one byte of data from the file.
close : a function of ifstream class that is used to close a file.
*/